src/reconcile/rec_row.c

```diff
@@ -460,11 +464,17 @@ __wti_rec_row_int(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
 
         /* Update compression state. */
         __rec_key_state_update(r, false);
+
+        /*
+         * Set the ref_changes state to zero if there were no concurrent changes while reconciling
+         * the internal page.
+         */
+        __wt_atomic_casv16(&ref->ref_changes, prev_ref_changes, 0);
```

Comment:
I don't think the fact that `prev_ref_changes` is equal to the old value is a reliable indicator that it wasn't changed. The ABA problem (https://en.wikipedia.org/wiki/ABA_problem) applies here.

For the prototype, I'm not sure whether we want to fix this. If we choose not to fix it, we should have a `TODO` comment + a ticket.

Fixing it is probably not hard - the ref is 16 bits, so we could extend it to have a reset counter in the upper 16 bits.

Reply:
ref_changes can only increase. Decrease can only be done by the reconciliation in a single thread so I don't think the ABA problem applies here.

Reply:
@quchenhao Will and I are wondering, can two reconciliation threads work on the same ref?

Reply:
No, they can't. This is because eviction needs exclusive access. If checkpoint is working on the page, eviction cannot, vice versa.

Reply:
Ok, I think this means that the ABA thing is not a problem.

Reply:
Thanks for confirming Chenhao - yeah, not having other threads able to reset the value makes this OK.